app/response_processor.py

```diff
@@ -1,132 +1,90 @@
+            raise BadMessageError("Missing ru_ref in metadata")
+        return
 
-        endpoint = host + "/" + path
+    def _decrypt(self, token, secret):
+        f = Fernet(secret)
+        try:
+            message = f.decrypt(token)
+        except TypeError:
+            message = f.decrypt(token.encode("utf-8"))
+        return message.decode("utf-8")
 
-        self.logger.debug("Sending receipt to CTP", endpoint=endpoint, tx_id=decrypted_json['tx_id'])
+    def _encode(self, decrypted):
+        if 'metadata' in decrypted and 'ru_ref' in decrypted['metadata']:
+            return {'caseRef': decrypted['metadata']['ru_ref']}
+        raise BadMessageError('Missing metadata')
 
-        receipt = {'caseRef': decrypted_json['metadata']['ru_ref']}
+    def _send_receipt(self, decrypted, data):
+        endpoint = settings.RECEIPT_HOST + "/" + settings.RECEIPT_PATH
+        if endpoint == "/":
+            raise BadMessageError("Unable to determine delivery endpoint from message")
 
-        response = self.remote_call(
-            endpoint,
-            json=receipt,
-            verify=False,
-            auth=(settings.RECEIPT_USER, settings.RECEIPT_PASS))
-        return self.response_ok(response)
+        headers = None
+        auth = (settings.RECEIPT_USER, settings.RECEIPT_PASS)
 
-    def remote_call(self, request_url, json=None, headers=None, verify=True, auth=None):
         try:
-            self.logger.info("Calling service", request_url=request_url)
-            r = None
-
-            if json:
-                r = session.post(request_url, json=json, headers=headers, verify=verify, auth=auth)
-            else:
-                r = session.get(request_url, headers=headers, verify=verify, auth=auth)
+            self.logger.info("Calling service", request_url=endpoint)
+            res = session.post(endpoint, data=data, headers=headers, verify=False, auth=auth)
 
-            return r
+            if res.status_code == 400:
+                self.logger.error("Receipt rejected by endpoint", request_url=res.url, status_code=400)
+                raise BadMessageError("Failure to send receipt")
 
-        except MaxRetryError:
-            self.logger.error("Max retries exceeded (5)", request_url=request_url)
+            elif res.status_code != 200 and res.status_code != 201:
+                # Endpoint may be temporarily down
+                self.logger.error("Bad response from endpoint", request_url=res.url, status_code=res.status_code)
+                raise RetryableError("Bad response from endpoint")
 
-    def response_ok(self, res):
-        if res.status_code == 200 or res.status_code == 201:
-            self.logger.info("Returned from service", request_url=res.url, status_code=res.status_code)
-            return True
+            else:
+                self.logger.info("Returned from service", request_url=res.url, status_code=res.status_code)
```

Comment:
Should already have it - gets bound to the logger in `_validate` once we know we have it (line 41)